packages/block-library/src/audio/edit.js

```diff
@@ -31,23 +30,22 @@ import { createUpgradedEmbedBlock } from '../embed/util';
 
 const ALLOWED_MEDIA_TYPES = [ 'audio' ];
 
-class AudioEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.toggleAttribute = this.toggleAttribute.bind( this );
-		this.onSelectURL = this.onSelectURL.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-	}
+function AudioEdit( {
+	attributes,
+	noticeOperations,
+	setAttributes,
+	onReplace,
+	isSelected,
+	noticeUI,
+} ) {
+	const { id, autoplay, caption, loop, preload, src } = attributes;
```

Comment:
Minor simplification (not really important):
```suggestion
function AudioEdit( {
	attributes: { id, autoplay, caption, loop, preload, src },
	noticeOperations,
	setAttributes,
	onReplace,
	isSelected,
	noticeUI,
} ) {
```

Reply:
`attributes` is used by `MediaPlaceholder` components this is why I decompose in two steps. With destructuration at "props" level linter is angry 😁

Reply:
Ah, I missed that. Thanks.